component library development

sarthaklav opened the pull request:
Worked on this component library and added various components to it.

sarthaklav committed:
added: some basic docs layout

sarthaklav committed:
add: avatar css is done

avatar css is done,
documentation will be done in next itration,
button has been started now

sarthaklav committed:
added: button style done

docs and enhancement will be done in another itration

sarthaklav committed:
adding : alert is on the way

sarthaklav committed:
added : basic working alert

alert working with setTimeout

sarthaklav committed:
added : 2 type of alerts are done

sarthaklav committed:
added: basic badges 1-1 on icon and avatar

sarthaklav committed:
refact: grid

(minor one line change)

sarthaklav committed:
feat: added 2 cards

sarthaklav committed:
refact : badges and some minor changes in other files

sarthaklav committed:
feat: added badge and text over card image

sarthaklav committed:
fix: fixed primary button width as color was not getting applied on border

sarthaklav committed:
feat : cards with other feature started

sarthaklav committed:
docs: added comments for alert in js

sarthaklav committed:
fact : cards done except js

dismiss button on card is yet to work

sarthaklav committed:
style: doc page structure improved

style is yet to be improved

sarthaklav committed:
feat: list started

sarthaklav committed:
feat: basic navbar started

sarthaklav committed:
feat: fonts added

sarthaklav committed:
feat: nav and sidebar imroved in doc page

sarthaklav committed:
feat: added list component

sarthaklav committed:
beautify docs.html

sarthaklav committed:
feat: added input/form and simplified-grid

sarthaklav committed:
feat : added text utilities

sarthaklav committed:
feat : added navbar style

sarthaklav committed:
feat : code added using carbon.now.sh

sarthaklav committed:
refact : badge and image

sarthaklav committed:
feat : added docs link in home page

sarthaklav committed:
Feat : added getting started in doc page

sarthaklav committed:
removed unnnecessary part and comments

sarthaklav committed:
deleted : unnecessary file

DanishOnGit commented on css-components/avatar.css:
```diff
@@ -0,0 +1,31 @@
+:root{
+    --extra-small-percent:5%;
+    --small-percent:10%;
+    --medium-percent:15%;
+    --large-percent:20%;
+}
```
Hi Sarthak, its a good practice to keep your :root in one place.

DanishOnGit commented on css-components/button.css:
```diff
@@ -0,0 +1,74 @@
+    border: var(--btn-border-width) solid var(--main-color);
+    margin: .5rem;
+}
+.link-primary{
+    background-color: var(--main-color);
+    color: var(--btn-color-light);
+    
+}
+.link-secondary{
+    color: var(--main-color);  
```
For link buttons you can check how Chakra or bootstrap has made their buttons, 
[chakra buttons](https://chakra-ui.com/docs/form/button#button-variants)

DanishOnGit commented on css-components/card.css:
```diff
@@ -0,0 +1,52 @@
+.card{
+   display: flex;
+   flex-direction: column;
+   /* justify-content: space-around; */
+   border: 1px solid var(--dark-gray-color);
+   width: var(--width-medium);
+   /* padding: var(--small-padding); */
```
You should remove the comments if not needed

DanishOnGit commented on css-components/card.css:
```diff
@@ -0,0 +1,52 @@
+   /* padding: var(--small-padding); */
+}
+
+.card .card-img{
+    width: 100%;
+}
```
Simply `.card-img` should also work. Let me know if i am missing anything

DanishOnGit commented on css-components/card.css:
```diff
@@ -0,0 +1,52 @@
+    width: 100%;
+}
+
+.card .card-details {
+    flex-grow: var(--small-number-unit);
+    margin: .5rem;
+}
```
same here as above

DanishOnGit commented on css-components/docs.css:
```diff
@@ -0,0 +1,148 @@
+    display: grid;
+    grid-template-rows: 100px 1fr 100px;
+    grid-template-columns: 1fr 5fr;
+    margin: var(--zero-percent);
```
You can set the margin using universal selector instead:
```
 *{
margin:0;
padding:0;
box-sizing:border-box
}```

DanishOnGit commented on css-components/docs.css:
```diff
@@ -0,0 +1,148 @@
+    height: 66vh;
+    position: fixed;
+    font-weight: 700;
+    z-index: 2;
+    top: 4rem;
```
its good to put z-indices too in variables

DanishOnGit commented on css-components/docs.css:
```diff
@@ -0,0 +1,148 @@
+.header {
+    grid-area: head;
+    background-color: grey;
+    /* background-color: var(--main-color); */
+    color: white;
+    /* position: fixed; */
```
remove comments wherever not needed

DanishOnGit commented on css-components/docs.css:
```diff
@@ -0,0 +1,148 @@
+    font-weight: 700;
+    z-index: 2;
+    top: 4rem;
+
+
+}
+
+.aside-link {
+    text-decoration: none;
+    color: black;
+
+}
```
formatting

DanishOnGit commented on css-components/docs.css:
```diff
@@ -0,0 +1,148 @@
+
+.aside-link:hover {
+    color: var(--link-hover-color);
+    /* background-color: black;
+    color: white; */
+    /* border-radius: .5rem;
+    border: .12rem solid black; */
```
comments should be removed

DanishOnGit commented on css-components/docs.css:
```diff
@@ -0,0 +1,148 @@
+    bottom: 0px;
+    background-color: var(--light-gray-color);
+}
+.footer >div{
+
+    margin: 1.5rem;
+}
+.footer ul{
+    display: flex;
+    flex-direction: row;
+    list-style-type: none;
+    justify-content: space-around;
+}
+.footer a{
+  color: black;
+}
```
Use classes to target elements instead. Though here , since your html markup won't change frequently, so not a big deal. But still, using classes is better.

DanishOnGit commented on css-components/docs.css:
```diff
@@ -0,0 +1,148 @@
+    margin-bottom: -1px;
+    height: 18vh;
+    width: 100%;
+    bottom: 0px;
```
`bottom` property will not have any effect on non-positioned elements. You can remove it.

DanishOnGit commented on css-components/docs.css:
```diff
@@ -0,0 +1,148 @@
+    margin-left: -1px;
+}
+
+h1 span{
+   color: var(--black-color);
+}
```
Use a class name instead. A good class name will help understand what is getting styled

DanishOnGit commented on css-components/input.css:
```diff
@@ -0,0 +1,29 @@
+    margin-bottom: 1rem;
+    height: 2rem;
+}
+.form-div-item-div{
+    /* margin-right: 20%; */
+}
```
This can be removed

DanishOnGit commented on css-components/navbar.css:
```diff
@@ -0,0 +1,55 @@
+    margin-right: auto;
+}
+
+.navbar-nav-list >li{
+    list-style-type: none;
+    margin-right: 1rem;
+}
+.navbar-nav-list >li>a{
+    font-size: 1rem;
+    font-weight: 500;
```
Avoid overly complex selectors. Using classes is better

DanishOnGit commented on css-components/root-styles.css:
```diff
@@ -0,0 +1,45 @@
+    --fifty-percent :50%;
+    
+}
+*{
+    box-sizing: border-box;
+}
```
`margin:0, padding:0`

DanishOnGit commented on css-components/text-utilities.css:
```diff
@@ -0,0 +1,43 @@
+    text-transform: uppercase;
+}
+.text-capitlized{
+    text-transform: capitalize!Important ;
```
I think this is a syntax issue here

DanishOnGit commented on css-components/toast.css:
```diff
@@ -0,0 +1,11 @@
+.toast{
+    display: none;
+    background-color: var(--dark-gray-color);
+    background-color: rgb(206, 7, 7);
```
You forgot a css var here

DanishOnGit commented on index.html:
```diff
@@ -0,0 +1,81 @@
+                    <a href="/docs/docs.html">Docs</a>
+                </li>
+            </ul>
+            <form action="" class="navbar-form">
+                <input class="nav-input" type="search" name="" id="" placeholder="">
+                <button class="btn btn-light submit-btn" type="submit">Search</button>
+            </form>
```
You can remove the unused attributes. Also a placeholder text like..."Search the docs" will be useful

DanishOnGit commented on index.html:
```diff
@@ -0,0 +1,81 @@
+
+    <main class="main">
+        <div class="main-content">
+            <img class="home-img" src="images/home-page-img.svg" alt="">
```
give an `alt` attribute for better accessibility

DanishOnGit commented on index.html:
```diff
@@ -0,0 +1,81 @@
+    <footer class="footer">
+        <div class="footer-content">
+            <ul>
+                <li> <a href="">github </a></li>
+                <li><a href="">twitter</a></li>
+                <li><a href="">linkedin</a></li>
```
I think you forgot to add the links

DanishOnGit reviewed (commented):
Hi Sarthak ! I have added the suggestions. Also you can rearrange the side bar on your website in alphabetical order to ease search.

neharajesh commented on css-components/modal.css:
```diff
@@ -0,0 +1,6 @@
+dialog{
+    z-index: 2;
```
Adding z-index as a root variable would help you maintain the count better!

neharajesh commented on css-components/docs.css:
```diff
@@ -0,0 +1,148 @@
+
+body{
+    display: grid;
+    grid-template-rows: 100px 1fr 100px;
```
Using rem here will help keep your site more responsive!

neharajesh reviewed (commented):
Hey Sarthak, here are my suggestions, hope they help!
1. Buttons on hover can show a pointer cursor.
2. Card headings could have a padding/margin on the left.
3. Height of the iframes could be adjusted accordingly.
4. Modal does not open, do check! 
5. Social links have to be added! 

Everything else looks good!

sarthaklav committed:
updated README

sarthaklav committed:
refactor style: body css

sarthaklav committed:
refact : removed unnecessary rating import

sarthaklav committed:
refact : changed root variables

changed root variables and placed at one place, gave values with variables, removed unused comments
